### PR TITLE
TLPORTAL-417-2 - if there is no corresponding Canvas course title, us…

### PR DESCRIPTION
**Pull request by gsilver**
…e a generic anchor

### CODE REVIEW

File: UI/data/todolms/ralt.json [hunk 1]
```diff
@@ -113,6 +113,18 @@
       "id": "assignment_57680",
       "grade_type": "points",
       "due_date_sort": "1454475540"
+    },{
```
Comment (dlhaines) [inline UI/data/todolms/ralt.json]: Is this test data?  If so can it be under a 'test' directory?

File: UI/js/controllers/controllers.js [hunk 2]
```diff
@@ -101,13 +100,18 @@ dashboardApp.controller('scheduleController', ['Schedule', '$scope', '$rootScope
     }];
 
    $scope.$on('canvasCourses', function (event, canvasCourses) {
+      //listen for changes to the Canvas course array (created by the Courses controller)
+      // and match the course title of the Canvas assignments to the course title in the array
       $.each($scope.schedule, function() {
         if(this.contextLMS === 'canvas'){
           var thisId = _.last(this.contextUrl.split('/'));
           var thisContext = _.findWhere(canvasCourses, {id: thisId});
```
Comment (dlhaines) [inline UI/js/controllers/controllers.js]: CanvasCourses is a pre-existing array?

Comment (gsilver) [inline UI/js/controllers/controllers.js]: It is an array that is created by the Courses controller. Which may finish its work before or after the  Schedule controller. $scope.$on('canvasCourses', <callback>)  is a watch for that array coming into existence and then  fires the callback  that modifies the data bound to the Schedule view - updating Canvas course titles.

File: UI/data/todolms/ralt.json (hunk 1 above)
Comment (gsilver) [inline UI/data/todolms/ralt.json]: Could - but schedule test data is perishable.

**Pull request merged**